### PR TITLE
Add Headless Chrome Support

### PR DESCRIPTION
**Pull request by mnifakram**
Add chrome
Use a template (Docker.yml) and a generator

### CODE REVIEW

File: dockerfile-generator.rb [hunk 1]
```diff
@@ -0,0 +1,392 @@
+#!/usr/bin/env ruby
```
Comment (NullVoxPopuli) [inline dockerfile-generator.rb:1]: there is a lot going on here -- what's the reason behind more than just `Dockerfile` additions?

Comment (mnifakram) [reply dockerfile-generator.rb:1]: it's just a generator to keep the `Dockerfile` more readable and well formatted and make it easier to add things by just adding them to the Dockerfile.yml

Comment (NullVoxPopuli) [reply dockerfile-generator.rb:1]: I guess, what's the point? 
Ya either add stuff to `Dockerfile` or `Dockerfile.yml`

The formatting can be handled manually, imo. I don't think someone should _have_ to have ruby installed to build any docker image. (docker is all about trying to have as few dependencies as possible)

**Comment by NullVoxPopuli** [comment]
I like the readme changes!

File: dockerfile-generator.rb (hunk 1 above)
Comment (mnifakram) [reply dockerfile-generator.rb:1]: They don't need any dependencies it's on the maintainer of this repo to add things to `Dockerfile.yml` and then run the script to generate the `Dockerfile` and people who want to use it they already have the `Dockerfile` pretty and ready to use.

Comment (mnifakram) [reply dockerfile-generator.rb:1]: It's like a linter to anyone that want to update the `Dockerfile` but they could still do it manually.

Comment (NullVoxPopuli) [reply dockerfile-generator.rb:1]: It just increases the barrier to entry for contributors.

Comment (mnifakram) [reply dockerfile-generator.rb:1]: I will add Contributors section at the end of the readme(not now though : ) ).

Comment (NullVoxPopuli) [reply dockerfile-generator.rb:1]: > It's like a linter to anyone that want to update the Dockerfile but they could still do it manually.
  
But then someone would need to update the template manually. imo, it's better to just have the one file.

Comment (mnifakram) [reply dockerfile-generator.rb:1]: yeah but updating the template it's more easier that the `Dockerfile` itself. It's more easy and  faster to add command or anything. At least from my experience when trying to build it this morning

Comment (NullVoxPopuli) [reply dockerfile-generator.rb:1]: I guess I don't understand?

can you describe the struggles you were having with the regular dockerfile?

Comment (mnifakram) [reply dockerfile-generator.rb:1]: keeping the same format and the `&&` and `\` at the end. if you want me to just update the `Dockerfile` I'm ok with that I can update it now.

**Pull request closed without merging**